crates/holochain/src/conductor/conductor.rs

```diff
@@ -849,21 +885,25 @@ where
     }
 
     /// Remove cells from the cell map in the Conductor
-    pub(super) async fn remove_cells(&mut self, cell_ids: Vec<CellId>) {
-        for cell_id in cell_ids {
-            if let Some(item) = self.cells.remove(&cell_id) {
-                // TODO: make this function async, or ensure that cleanup is fast,
-                //       so we don't hold the conductor lock unduly long while doing cleanup
-                //       (currently cleanup is a noop, so it's not urgent)
-                if let Err(err) = item.cell.cleanup().await {
-                    tracing::error!("Error cleaning up Cell: {:?}\nCellId: {}", err, cell_id);
-                }
+    pub(super) async fn remove_cells(&self, cell_ids: Vec<CellId>) {
+        let to_cleanup: Vec<_> = self.cells.share_mut(|cells| {
+            cell_ids
+                .into_iter()
+                .filter_map(|cell_id| cells.remove(&cell_id).map(|c| (cell_id, c)))
+                .collect()
+        });
+        for (cell_id, item) in to_cleanup {
+            // TODO: make this function async, or ensure that cleanup is fast,
+            //       so we don't hold the conductor lock unduly long while doing cleanup
+            //       (currently cleanup is a noop, so it's not urgent)
```

Comment:
Is this comment irrelevant now, since we are not holding the conductor lock anymore?

Reply:
Yeah I think that's right

```suggestion
```